ntp/syncTime.go

```diff
@@ -119,35 +118,33 @@ func (s *syncTime) StartSync() {
 // sync method does the time synchronization and sets the current offset difference between local time
 // and server time with which it has done the synchronization
 func (s *syncTime) sync() {
-	if s.query != nil {
-		clockOffsetSum := time.Duration(0)
-		succeededRequests := 0
+	clockOffsetSum := time.Duration(0)
+	succeededRequests := 0
 
-		for i := 0; i < totalRequests; i++ {
-			r, err := s.query(s.ntpOptions)
+	for i := 0; i < len(s.ntpOptions.Host); i++ {
+		r, err := s.query(s.ntpOptions)
 
-			if err != nil {
-				log.Error(fmt.Sprintf("NTP Error: %s", err))
-				//Change host if current host return an error
-				newHostIndex := checkNTPHost(config.NTPConfig{
-					Host: s.ntpOptions.Host,
-					Port: s.ntpOptions.Port,
-				}, s.query)
-				s.ntpOptions.HostIndex = newHostIndex
+		if err != nil {
+			log.Error(fmt.Sprintf("NTP Error: %s", err))
+			//Change host if the current host returns an error
+			newHostIndex := checkNTPHost(config.NTPConfig{
+				Hosts: s.ntpOptions.Host,
+				Port:  s.ntpOptions.Port,
+			}, s.query)
+			s.ntpOptions.HostIndex = newHostIndex
 
-				continue
-			}
+			continue
+		}
 
-			log.Debug(fmt.Sprintf("NTP reading: %s", r.Time.Format("Mon Jan 2 15:04:05 MST 2006")))
+		log.Debug(fmt.Sprintf("NTP reading: %s", r.Time.Format("Mon Jan 2 15:04:05 MST 2006")))
```

Comment:
also here, lowercase log message

Reply:
in all the cases are used NPT with upper letters, i will not change it.